pkg/scheduler/util/utils.go

```diff
@@ -109,3 +109,39 @@ func GetPodAntiAffinityTerms(podAntiAffinity *v1.PodAntiAffinity) (terms []v1.Po
 	}
 	return terms
 }
+
+// GetZoneKey is a helper function that builds a string identifier that is unique per failure-zone;
+// it returns empty-string for no zone.
+// Since there are currently two separate zone keys:
+//   * "failure-domain.beta.kubernetes.io/zone"
+//   * "topology.kubernetes.io/zone"
+// GetZoneKey will first check failure-domain.beta.kubernetes.io/zone and if not exists, will then check
+// topology.kubernetes.io/zone
+// NOTE: This function copied from pkg/util/node package to avoid external scheduler dependency
+func GetZoneKey(node *v1.Node) string {
+	labels := node.Labels
+	if labels == nil {
+		return ""
+	}
+
+	// TODO: prefer stable labels for zone in v1.18
```

Comment:
can just look at the stable labels. a 1.19 scheduler can depend on a 1.18+ controller, which ensures the stable labels exist and are synced with the beta labels.